test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJcmd.java

```diff
@@ -163,6 +172,48 @@ public void testClassHistogramAll() throws IOException {
 		log(EXPECTED_STRING_FOUND);
 	}
 
+	@Test
+	public void testDumps() throws IOException {
+		@SuppressWarnings("nls")
+		String[][] commandsAndDumpTypes = {{DUMP_HEAP, "Heap"}, {GC_HEAP_DUMP, "Heap"}, {DUMP_JAVA, "Java"}, {DUMP_SNAP, "Snap"}, {DUMP_SYSTEM, "System"}};
+		for (String[] commandAndDumpName: commandsAndDumpTypes) {
+			TargetManager tgt = new TargetManager(TestConstants.TARGET_VM_CLASS, null,
+					Collections.emptyList(), Collections.singletonList("-Xmx10M")); //$NON-NLS-1$
+			tgt.syncWithTarget();
+			String targetId = tgt.targetId;
+			assertNotNull(targetId, "target did not launch"); //$NON-NLS-1$
+			String dumpType = commandAndDumpName[1];
+			@SuppressWarnings("nls")
+			File dumpFileLocation = new File(userDir, "my" + dumpType + "Dump");
+			dumpFileLocation.delete();
+			try {
+				String dumpFilePath = dumpFileLocation.getAbsolutePath();
+				List<String> args = new ArrayList<>();
+				args.add(targetId);
+				String command = commandAndDumpName[0];
+				log("test " + command); //$NON-NLS-1$
+				args.add(command);
+				args.add(dumpFilePath);
+				List<String> jcmdOutput = runCommandAndLogOutput(args);
+
+				assertTrue(dumpFileLocation.exists(), "dump file " + dumpFilePath + "missing");  //$NON-NLS-1$//$NON-NLS-2$
+				String expectedString = "Dump written to " + dumpFilePath; //$NON-NLS-1$
+				log("Expected jcmd output string: " + expectedString); //$NON-NLS-1$
+				Optional<String> searchResult = StringUtilities.searchSubstring(expectedString, jcmdOutput);
+				assertTrue(searchResult.isPresent(), "Expected string not found in jcmd output: " + expectedString); //$NON-NLS-1$
+				log(EXPECTED_STRING_FOUND + " in jcmd output: " + expectedString); //$NON-NLS-1$
+
+				expectedString = dumpType + " dump written to " + dumpFilePath; //$NON-NLS-1$
+				searchResult = StringUtilities.searchSubstring(expectedString,tgt.getTargetErrReader().lines());
+				assertTrue(searchResult.isPresent(), "Expected string not found in target standard error: " + expectedString); //$NON-NLS-1$
+				log(EXPECTED_STRING_FOUND + " in target standard error: " + expectedString); //$NON-NLS-1$
+			} finally {
+				tgt.terminateTarget();
+				dumpFileLocation.delete();
```

Comment:
Is this redundant since same call is made right after `new File(userDir, "my" + dumpType + "Dump")`?

Reply:
Additional question, is it required to delete the folder `dumpFileLocation` for each sub-test? Can all sub-tests share same dump file folder? Can the `TargetManager tgt` be re-used?

Reply:
> Is this redundant since same call is made right after `new File(userDir, "my" + dumpType + "Dump")`?

If a previous run did not clean up, e.g. due to a crash, I still want to make sure the directory is clean, since the test will fail if the requested file already exists.


> Additional question, is it required to delete the folder `dumpFileLocation` for each sub-test? 

Yes: the test will fail if the file already exists, since in that case the dump agent chooses another file in another directory. 

> Can all sub-tests share same dump file folder? 

They already do.

> Can the `TargetManager tgt` be re-used?

Possibly, though it makes the code more complicated (extra `try/finally`).  Launching and terminating the target is quick.